src/node_crypto.cc

```diff
@@ -3156,12 +3210,15 @@ void CipherBase::SetAuthTag(const FunctionCallbackInfo<Value>& args) {
 
   // Restrict GCM tag lengths according to NIST 800-38d, page 9.
   unsigned int tag_len = Buffer::Length(args[0]);
-  if (tag_len > 16 || (tag_len < 12 && tag_len != 8 && tag_len != 4)) {
-    char msg[125];
-    snprintf(msg, sizeof(msg),
-             "Permitting authentication tag lengths of %u bytes is deprecated. "
-             "Valid GCM tag lengths are 4, 8, 12, 13, 14, 15, 16.", tag_len);
-    ProcessEmitDeprecationWarning(cipher->env(), msg, "DEP0090");
+  const int mode = EVP_CIPHER_CTX_mode(cipher->ctx_);
+  if (mode == EVP_CIPH_GCM_MODE) {
+    if (tag_len > 16 || (tag_len < 12 && tag_len != 8 && tag_len != 4)) {
+      char msg[125];
+      snprintf(msg, sizeof(msg),
+          "Permitting authentication tag lengths of %u bytes is deprecated. "
+          "Valid GCM tag lengths are 4, 8, 12, 13, 14, 15, 16.", tag_len);
+      ProcessEmitDeprecationWarning(cipher->env(), msg, "DEP0090");
+    }
```

Comment:
I think we should add the check of CCM auth tag not be less than 64 bytes according to NIST800 38c

Reply:
The very same section of that publication contains an example of an application for which an authentication tag length of 32 bits is enough. The CCM implementatin is much less likely to accept inauthentic messages than the current GCM implementation which accepts single-byte tags, and - contrary to GCM - users need to explicitely specify the CCM tag length (assuming they don't simply use `authTagLength: receivedTag.length`).

38d contains similiar specifications for the GCM tag length, which usually should be at least 96 bits, but it explicitely permits shorter tags "for certain applications".

Reply:
Okay, the doc of crypto module notes user's responsibility for selecting the crypto algorithm and key size. So we need not to limit its tag length.